paasta_tools/setup_prometheus_adapter_config.py

```diff
@@ -453,8 +450,14 @@ def create_instance_uwsgi_scaling_rule(
             )[{moving_average_window}s:]
         )
     """
+
+    # our Prometheus query is calculating a desired number of replicas, and then k8s wants that expressed as an average utilization
+    # so as long as we divide by the number that k8s ends up multiplying by, we should be able to convince k8s to run any arbitrary
+    # number of replicas.
+    # k8s happens to multiply by the # of ready pods - so we divide by that rather than by the amount of current replicas (which may
+    # include non-ready pods)
```

Comment:
might want to link to the k8s source line in question https://github.com/kubernetes/kubernetes/blob/7ec1a89a509906dad9fd6a4635d7bfc157b47790/pkg/controller/podautoscaler/replica_calculator.go#L278

Reply:
done :)